File: src/main/java/com/brasil/transparente/api/util/OrdererService.java
package com.brasil.transparente.api.util;

import com.brasil.transparente.api.entity.*;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Objects;

@Service
@NoArgsConstructor
public class OrdererService {

    public void orderDespesaSimplificadaListBySpending(List<DespesaSimplificada> despesaSimplicadaList) {
        despesaSimplicadaList.sort((i1, i2) -> Double.compare(i2.getTotalValue(), i1.getTotalValue()));
        for (int i = 0; i < despesaSimplicadaList.size(); i++) {
            if (Objects.equals(despesaSimplicadaList.get(i).getName(), "Outros")) {
                DespesaSimplificada outros = despesaSimplicadaList.remove(i);
                despesaSimplicadaList.add(outros);
                break;
            }
        }
    }


}

File: src/main/kotlin/com/brasil/transparente/api/util/OrdererService.kt
package com.brasil.transparente.api.util

import com.brasil.transparente.api.entity.DespesaSimplificada
import org.springframework.stereotype.Service

@Service
class OrdererService {

    companion object {
        private const val OUTROS = "Outros"
    }

    fun orderDespesaSimplificadaListBySpending(despesaSimplificadaList: MutableList<DespesaSimplificada>) {
            despesaSimplificadaList.sortByDescending { it.totalValue }
            val outros = despesaSimplificadaList.firstOrNull { it.name == OUTROS}
            if (outros != null) {
                despesaSimplificadaList.remove(outros)
                despesaSimplificadaList.add(outros)
            }
    }

}